webapi/Controllers/ChatHistoryController.cs

```diff
@@ -240,4 +241,96 @@ public async Task<IActionResult> GetAllChatSessionsAsync(string userId)
 
         return this.NotFound($"No chat session found for chat id '{chatId}'.");
     }
+
+    /// <summary>
+    /// Delete a chat session.
+    /// </summary>
+    /// <param name="userId">Unique Id of user who initiated the request.</param>
+    /// <param name="sessionId">The chat id.</param>
+    [HttpDelete]
+    [Route("chatSession/{sessionId:guid}")]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> DeleteChatSessionAsync([FromServices] IHubContext<MessageRelayHub> messageRelayHubContext, [FromBody] DeleteChatRequest requestBody, Guid sessionId)
+    {
+        var chatId = sessionId.ToString();
+        var userId = requestBody.UserId;
+
+        if (userId == null)
+        {
+            return this.BadRequest("UserId cannot be null.");
+        }
+
+        ChatSession? chatToDelete = null;
+        try
+        {
+            // Make sure the chat session exists
+            chatToDelete = await this._sessionRepository.FindByIdAsync(chatId);
```

Comment:
We probably want to validate that the user is part of the chat before proceeding. Right now, the service will take the delete request from anyone even though they are not part of the chat.

Reply:
Ahh, just saw your description. Resolving this one.